### PR TITLE
Some improvements for CUDAWrappers::SparseMatrix

### PR DESCRIPTION
**Pull request by masterleinad**
There are new methods `CUDAWrappers::SparseMatrix::print()` and `CUDAWrappers::SparseMatrix::print_formatted()`. Additionally, `CUDAWrappers::SparseMatrix` gained a move assignment operator.

### CODE REVIEW

File: include/deal.II/lac/cuda_sparse_matrix.h [hunk 1]
```diff
@@ -136,6 +150,51 @@ namespace CUDAWrappers
...
+
+    /**
+     * Print the matrix to the given stream, using the format <tt>(row,column)
+     * value</tt>, i.e. one nonzero entry of the matrix per line. If
+     * <tt>across</tt> is true, print all entries on a single line, using the
+     * format row,column:value.
+     *
+     * If the argument <tt>diagonal_first</tt> is true, diagonal elements of
+     * quadratic matrices are printed first in their row, corresponding to the
+     * internal storage scheme. If it is false, the elements in a row are
```
Comment (Rombur) [inline include/deal.II/lac/cuda_sparse_matrix.h]: This is not how it is stored internally. The deal.II matrix is reordered to match cuSPARSE format.

File: include/deal.II/lac/cuda_sparse_matrix.h [hunk 2]
```diff
@@ -340,6 +399,131 @@ namespace CUDAWrappers
   {
     return nnz;
   }
+
+
+
+  template <typename Number>
+  template <class StreamType>
+  inline void
+  SparseMatrix<Number>::print(StreamType &out,
```
Comment (Rombur) [inline include/deal.II/lac/cuda_sparse_matrix.h]: Have you checked that the function prints the matrix correctly?

Comment (masterleinad) [reply include/deal.II/lac/cuda_sparse_matrix.h]: Yes, but let me write a test.

**Comment by masterleinad** [comment]
@Rombur Admittedly, I had to fix `CUDAWrappers::print` and its documentation indeed. I also added some tests. Please have another look.

**Pull request merged**